File: frontend/src/api/Student/attendance.ts
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000/api/student';

export interface AttendanceRecord {
  attendance_id: number;
  offering_id: number;
  student_id: string;
  attendance_date: string;
  status: string;
}

export const getAttendanceRecords = async (studentId: string, offeringId?: number): Promise<AttendanceRecord[]> => {
  try {
    const url = offeringId 
      ? `${API_BASE_URL}/students/${studentId}/attendance?offering_id=${offeringId}`
      : `${API_BASE_URL}/students/${studentId}/attendance`;
    const response = await axios.get<AttendanceRecord[]>(url);
    return response.data;
  } catch (error) {
    console.error("Error fetching attendance records:", error);
    throw error;
  }
}; 

File: backend/routers/student/attendance.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from crud.student import attendance as crud_attendance
from database import get_db
from typing import List, Optional
from pydantic import BaseModel
from datetime import date

class AttendanceRecordResponse(BaseModel):
    attendance_id: int
    offering_id: int
    student_id: str
    attendance_date: date
    status: str

    class Config:
        from_attributes = True

router = APIRouter()

@router.get("/students/{student_id}/attendance", response_model=List[AttendanceRecordResponse])
def read_attendance_records(student_id: str, offering_id: Optional[int] = None, db: Session = Depends(get_db)):
    attendance_records = crud_attendance.get_attendance_records(db, student_id=student_id, offering_id=offering_id)
    return attendance_records 